fda/fdata_boxplot.py

```diff
@@ -0,0 +1,572 @@
+"""Functional Data Boxplot Module.
+
+This module contains the classes to construct the functional data boxplot and
+visualize it.
+
+"""
+import matplotlib
+import matplotlib.pyplot as plt
+import math
+
+from .depth_measures import *
+from .grid import FDataGrid
+from io import BytesIO
+from abc import ABC, abstractmethod
+
+__author__ = "Amanda Hernando Bernabé"
+__email__ = "amanda.hernando@estudiante.uam.es"
+
+class FDataBoxplot(ABC):
+    """Abstract class inherited by the Boxplot and SurfaceBoxplot classes.
+
+    It the data of the functional boxplot or surface boxplot of a FDataGrid object,
+    depending on the dimensions of the domain, 1 or 2 respectively.
+
+    It forces to both classes, Boxplot and SurfaceBoxplot to conain at least the median,
+    central and outlying envelopes and a colormap for their graphical representation,
+    obtained calling the plot method.
+
+    """
+    @abstractmethod
+    def __init__(self, factor=1.5):
+        if factor < 0:
```

Comment:
Maybe factor should be stored, and exposed as a read-only property.

Reply:
done